controller/callbackcontroller.php

```diff
@@ -373,26 +373,29 @@ public function track($doc, $users, $key, $status, $url, $token) {
                 }
 
                 try {
+                    $token = isset($hashData->token) ? $hashData->token : NULL;
+                    $ownerId = $hashData->ownerId;
+                    if (empty($ownerId)) {
+                        $this->logger->error("Track without owner: " . $fileId . " status " . $trackerStatus, array("app" => $this->appName));
+                        return new JSONResponse(["message" => $this->trans->t("File owner is empty")], Http::STATUS_BAD_REQUEST);
+                    }
+
                     $userId = $users[0];
                     $user = $this->userManager->get($userId);
                     if (!empty($user)) {
-                        \OC_Util::tearDownFS();
-                        \OC_Util::setupFS($userId);
-
+                        // Login user to allow audit
                         $this->userSession->setUser($user);
-                    } else {
-                        $ownerId = $hashData->ownerId;
-
-                        \OC_Util::tearDownFS();
-                        if (!empty($ownerId)) {
-                            \OC_Util::setupFS($ownerId);
-                        }
                     }
 
-                    $token = isset($hashData->token) ? $hashData->token : NULL;
-                    list ($file, $error) = empty($token) ? $this->getFile($userId, $fileId) : $this->getFileByToken($fileId, $token);
+                    // Setup fs of file owner
+                    \OC_Util::tearDownFS();
+                    \OC_Util::setupFS($ownerId);
```

Comment:
FS however should be of file owner

Reply:
@LinneyS please remember that in OC, there are several layers. One is authentication, and other is FS access - these are completely independent, and app logic controls it.

Reply:
These fixes really helped me. We will try to re-test various opening cases of the editor and check the save.